nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-2-6-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/CustomKerberosLogin.java

```diff
@@ -0,0 +1,245 @@
+                } catch (InterruptedException ie) {
+                    log.warn("[Principal={}]: TGT renewal thread has been interrupted and will exit.", principal);
+                    return;
+                }
+                try {
+                    checkTGTAndReLogin();
+                } catch (Throwable t) {
+                    log.error("[Principal={}]: Error from TGT refresh thread", principal, t);
+                }
+            }
+        });
+        t.start();
+        return loginContext;
+    }
+
+    @Override
+    public void close() {
+        if ((t != null) && (t.isAlive())) {
+            t.interrupt();
+            try {
+                t.join();
+            } catch (InterruptedException e) {
+                log.warn("[Principal={}]: Error while waiting for Login thread to shutdown.", principal, e);
+                Thread.currentThread().interrupt();
+            }
+        }
+    }
+
+    @Override
+    public Subject subject() {
+        return subject;
+    }
+
+    @Override
+    public String serviceName() {
+        return serviceName;
+    }
+
+    private synchronized void checkTGTAndReLogin() throws LoginException {
+        final KerberosTicket tgt = getTGT();
+        if (tgt == null) {
+            log.info("[Principal={}]: TGT was not found, performing login", principal);
+            reLogin();
+            return;
+        }
+
+        if (System.currentTimeMillis() < getRefreshTime(tgt)) {
+            log.debug("[Principal={}]: TGT was found, but has not reached expiration window", principal);
+            return;
+        }
+
+        try {
+            tgt.refresh();
+            log.info("[Principal={}]: TGT refreshed", principal);
+            getRefreshTime(tgt);
+        } catch (RefreshFailedException rfe) {
+            log.warn("[Principal={}]: TGT refresh failed, will attempt relogin", principal);
+            log.debug("", rfe);
+            reLogin();
+        }
+    }
+
+    private static String getServiceName(Map<String, ?> configs, String contextName, Configuration configuration) {
+        List<AppConfigurationEntry> configEntries = Arrays.asList(configuration.getAppConfigurationEntry(contextName));
+        String jaasServiceName = JaasContext.configEntryOption(configEntries, JaasUtils.SERVICE_NAME, null);
+        String configServiceName = (String) configs.get(SaslConfigs.SASL_KERBEROS_SERVICE_NAME);
+        if (jaasServiceName != null && configServiceName != null && !jaasServiceName.equals(configServiceName)) {
+            String message = String.format("Conflicting serviceName values found in JAAS and Kafka configs " +
+                    "value in JAAS file %s, value in Kafka config %s", jaasServiceName, configServiceName);
+            throw new IllegalArgumentException(message);
+        }
+
+        if (jaasServiceName != null)
+            return jaasServiceName;
+        if (configServiceName != null)
+            return configServiceName;
```

Comment:
Recommend using standard brackets as opposed to shortened conditional form.
```suggestion
        if (jaasServiceName != null) {
            return jaasServiceName;
        }
        if (configServiceName != null) {
            return configServiceName;
        }    
```